code/modules/research/reward_system/security.dm

```diff
@@ -0,0 +1,35 @@
+/obj/machinery/computer/department_reward/security
+	name = "departmental research console (Security)"
+	desc = "A primitive version of the famous R&D console used to unlock department-relevant research. This one is powered by incarcerated criminals."
+	department_display = "Security"
+	circuit = /obj/item/circuitboard/computer/department_reward/security
+
+/obj/machinery/computer/department_reward/security/Initialize(mapload)
+	nodes_available = GLOB.security_nodes
+	. = ..()
+
+/obj/machinery/computer/department_reward/security/check_reward(delta_time)
+	. = 0
+
+	// Criminals in brig cells with an active timer
+	var/list/criminals = list()
+	for(var/obj/machinery/door_timer/timer in SSmachines.processing)
```

Comment:
could always make a GLOB for this if you're worried about iterating through like 1000 machines